### PR TITLE
Split contribution doc into a separate file

### PR DESCRIPTION
**Pull request by paulfantom**
## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

Slowly implementing https://github.com/prometheus-operator/kube-prometheus/issues/1005

This PR moves contributing section into a separate file that is then enhanced with data from similar file in prometheus-operator repository.

## Type of change

_What type of changes does your code introduce to the kube-prometheus? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [x] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. Later this will be copied to the changelog file._

```release-note

```

### CODE REVIEW

**Review by PhilipGough — APPROVED** [review]
lgtm

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+Origin (DCO). This document was created by the Linux Kernel community and is a
+simple statement that you, as a contributor, have the legal right to make the
+contribution. See the [DCO](DCO) file for details.
+
+## Email and Chat
+
+The project currently uses the [Kubernetes Slack](https://slack.k8s.io/):
+- [#prometheus-operator](https://kubernetes.slack.com/archives/CFFDS2Z7F)
+- [#prometheus-operator-dev](https://kubernetes.slack.com/archives/C01B03QCSMN)
+
```
Comment (PhilipGough) [inline CONTRIBUTING.md:23]: @paulfantom worth mentioning https://github.com/prometheus-operator/kube-prometheus/discussions here?

Comment (paulfantom) [reply CONTRIBUTING.md:23]: Yes, good catch!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+Please avoid emailing maintainers directly.
+
+## Getting Started
+
+- Fork the repository on GitHub
+- Read the [README](README.md) for build and test instructions
+- Play with the project, submit bugs, submit patches!
```
Comment (ArthurSens) [inline CONTRIBUTING.md:30]:
```suggestion
- Play with the project, submit bug fixes, submit patches!
```

please don't submit bugs 🤣

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+## Contribution Flow
+
+This is a rough outline of what a contributor's workflow looks like:
+
+- Create a topic branch from where you want to base your work (usually `main`).
+- Make commits of logical units.
+- Make sure your commit messages are in the proper format (see below).
+- Push your changes to a topic branch in your fork of the repository.
+- Make sure the tests pass, and add any new tests as appropriate.
+- Submit a pull request to the original repository.
```
Comment (ArthurSens) [inline CONTRIBUTING.md:41]: Maybe it is worth mentioning how to execute tests? Just mentioning the Makefile step should be enough I think

Comment (ArthurSens) [reply CONTRIBUTING.md:41]: I don't want to sound like a sellout of my employer, but the `developer-workspace` folder was created as an alternative for contributors who don't have a good local environment set up. Do you think it is worth mentioning it?

**Comment by ArthurSens** [comment]
Regarding #1005, I was gonna mention that on today's contributor office hour but we didn't get to that point.

It would be nice if we could split the effort into smaller and well-defined issues and solve them one by one. I'm proposing this idea just because I'd like to help but I'm don't have a clear understanding of what we want :)

**Comment by paulfantom** [comment]
> It would be nice if we could split the effort into smaller and well-defined issues and solve them one by one. I'm proposing this idea just because I'd like to help but I'm don't have a clear understanding of what we want :)

The overall idea is that we should split README.md into parts so it will be easier to navigate on the website. In short, we want to move the following sections into separate documents living in `Documentation` directory (potentially more than one document from one section):
- [troubleshooting](https://github.com/prometheus-operator/kube-prometheus#troubleshooting)
- [customization examples](https://github.com/prometheus-operator/kube-prometheus#customization-examples)
- [Minikube Example](https://github.com/prometheus-operator/kube-prometheus#minikube-example) - this one might be a good candidate to add in contributing doc, but I wanted to start small
- [Continuous Delivery](https://github.com/prometheus-operator/kube-prometheus#continuous-delivery)

IMHO in readme we should have only the following content and use links to dive into other documents:
- Project Name
- (optional) Status Badges
- Project Description (what project does, what technologies are used, what are the challenges project tries to solve)
- Table Of Contents (used as an entry point to link to other docs in `Documentation/` directory)
- How to Install
- How to Use (brief overview with links to more in-depth scenarios)
- License link

File: CONTRIBUTING.md (hunk 3 above)
Comment (paulfantom) [reply CONTRIBUTING.md:41]: Yes, totally! Would like to write it in a follow-up PR? I think you have the most knowledge of what should be put here so people use `developer-workspace` and gtipod :)

File: CONTRIBUTING.md (hunk 1 above)
Comment (paulfantom) [reply CONTRIBUTING.md:29]: I have rewrote this section and modeled it after prometheus site - https://prometheus.io/community/

File: CONTRIBUTING.md (hunk 3 above)
Comment (ArthurSens) [reply CONTRIBUTING.md:58]: yep, sounds good to me!

**Review by ArthurSens — APPROVED** [review]
LGTM!

**Pull request merged**